projects/sandbox/train/train/data_structures.py

```diff
@@ -71,3 +74,65 @@ def __next__(self) -> Tuple[torch.Tensor, torch.Tensor]:
         if self.preprocessor is not None:
             X, y = self.preprocessor(X, y)
         return X, y
+
+
+class BBHNetWaveformInjection(RandomWaveformInjection):
+    def forward(self, X: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
+        if not self.training:
+            return X, y
+
+        X, idx, _ = super().forward(X)
+        y[idx] = 1
+        return X, y
+
+
+# TODO: generalize to arbitrary ifos
+class GlitchSampler(Transform):
```

Comment:
I think you can get rid of the `transforms` module and have this inherit directly from `torch.nn.Module`. Below for the parameters I would switch from `**glitches` to `*glitches` since we don't use the IFO name information and this makes it easier to make sure things are passed in the correct order (e.g. a user couldn't actually use the `**` syntax when initializing since they can't guarantee the dict they're using will maintain the order they want). Then for registering the params you could something like
```python
super().__init__()
for i, glitch in enumerate(glitches):
    glitch = torch.Tensor(glitch)
    self.register_buffer(f"glitch_{i}", glitch)
```
Then during `forward` you can do
```python
for i, glitch in enumerate(self.buffers()):
    ...
```

The reality is that we should probably be treating the glitches as `Tensor`s and keeping them on CPU then moving the sampled glitches to GPU before insertion, not `Buffer`s or `Parameter`s which get moved to whatever device the layer is on, but our glitch datatsets are small enough right now that this shouldn't be that big of a deal.

Reply:
You can also get rid of the `TODO` since this is now fully general to any number of IFOs

Reply:
Actually now that I'm seeing how this actually gets initialized below, I'm realizing that this would require specifying everything as args, e.g. `GlitchSampler(glitch_prob, max_offset, h1_glitches, l1_glitches)`, which is maybe less clear than allowing the user to specify `prob=..., max_offset=...`, so maybe just keep this for now and we can figure out a better way of doing this down the line.

Either way I would say drop the `transform` module and just switch the loop above to be
```python
for ifo, glitch in glitches.items():
    self.register_buffer(ifo, torch.Tensor(glitch))
```